autometa/common/external/prodigal.py

```diff
@@ -121,41 +125,50 @@ def run(assembly, nucls_out, prots_out, force=False,cpus=0,parallel=True):
     cmd = [str(arg) for arg in cmd]
     logger.debug(f'cmd: {" ".join(cmd)}')
     if parallel:
-        returncode = subprocess.call(" ".join(cmd), shell=True)
-        tmpfpaths = glob(os.path.join(tmpdir,'*.faa'))
-        lines = ''
-        for fp in tmpfpaths:
-            with open(fp) as fh:
-                for line in fh:
-                    lines += line
-        out = open(prots_out, 'w')
-        out.write(lines)
-        out.close()
-        tmpfpaths = glob(os.path.join(tmpdir, '*.fna'))
-        lines = ''
-        for fp in tmpfpaths:
-            with open(fp) as fh:
-                for line in fh:
-                    lines += line
-        out = open(nucls_out, 'w')
-        out.write(lines)
-        out.close()
-        shutil.rmtree(tmpdir)
+        try:
+            returncode = subprocess.call(" ".join(cmd), shell=True)
+            tmpfpaths = glob(os.path.join(tmpdir,'*.faa'))
+            lines = ''
+            for fp in tmpfpaths:
+                with open(fp) as fh:
+                    for line in fh:
+                        lines += line
+            out = open(prots_out, 'w')
+            out.write(lines)
+            out.close()
+            tmpfpaths = glob(os.path.join(tmpdir, '*.fna'))
+            lines = ''
+            for fp in tmpfpaths:
+                with open(fp) as fh:
+                    for line in fh:
+                        lines += line
+            out = open(nucls_out, 'w')
+            out.write(lines)
+            out.close()
+        except Exception as err:
+            # COMBAK: Should probably be more descriptive as to what errors could occur here.
+            logger.exception(err)
+        finally:
+            shutil.rmtree(tmpdir)
     else:
         with open(os.devnull, 'w') as stdout, open(os.devnull, 'w') as stderr:
             proc = subprocess.run(cmd, stdout=stdout, stderr=stderr)
             returncode = proc.returncode
     if returncode:
         logger.warning(f'Args:{cmd} ReturnCode:{returncode}')
+        # COMBAK: Check all possible return codes for GNU parallel
     for fp in [nucls_out, prots_out]:
         if not os.path.exists(fp):
```

Comment:
See issue #16, point 4. Perhaps BioPython could be used to check that `nucls_out` and `prots_out` are valid fasta files?

Reply:
I don't think there is a `validFasta` or some equivalent function in BioPython, but we can retrieve the records from the files with Biopython and if we do not encounter an error, assume they are valid.

Reply:
Yes, that is probably the next best thing, but note that it might not give an error with a zero size file (I guess that would technically be a valid fasta file with no contents). So perhaps check that `size != 0` and try loading with BioPython?